### PR TITLE
feat: dialog for uploading payment slip

### PR DESCRIPTION
**Pull request by natTP**
### What was done:

- [x] Implement dialog UI
- [x] Handle change `shirtSize`
- [x] Handle upload `payment` document

Dialog UI:
image

https://user-images.githubusercontent.com/57129145/149629068-fcc9d0d5-1aa7-4ca4-b31e-d9222cb36b06.mp4

Dialog Behavior:
- When dialog is opened, the currently selected `shirtSize` will be shown as selected. (default is none)
- If no `shirtSize` is selected, close button is disabled. (Can still close by clicking outside though.) If `shirtSize` is selected, the button will become enabled.
- `shirtSize` is sent to backend on dialog close, but only when the `shirtSize` was changed compared to before the dialog was opened.
- If a file is uploaded, will show preview.
- File size limit of 4MB
- Shows error message when file size limit exceeded, or upload timeout

### CODE REVIEW

**Comment by natTP** [comment]
Note: it seems that backend doesn't allow uploading files at any application state further than `SUBMITTED` yet. 
This dialog displays at state `PASSED_INTERVIEW`, so there may be an error.

**Review by miello — COMMENTED** [review]
IMO
1) We might need to add close button (Maybe CrossIcon) in modal because user might don't know how to close modal (instead of clickaway)
2) Should file uploaded be required instead of optional ?

File: src/modules/profile.module.tsx [hunk 1]
```diff
@@ -71,6 +77,12 @@ const ProfileModule = () => {
     <>
       <div className={classes.bg} />
       <Container maxWidth="lg" className={classes.container}>
+        <CustomDialog
+          open={isOpen}
+          existingShirtSize={application.shirtSize === null ? ("" as ValidShirtSize) : (application.shirtSize as ValidShirtSize)}
```
Comment (saenyakorn) [inline src/modules/profile.module.tsx]: You can use enum instead to avoid type casting

**Comment by saenyakorn** [comment]
Did you compress the QR? Too big picture cause longer loading.

**Comment by natTP** [comment]
> 1. We might need to add close button (Maybe CrossIcon) in modal because user might don't know how to close modal (instead of clickaway)
> 2. Should file uploaded be required instead of optional ?

File upload is optional because the dialog can also select shirt size, so users should be able to select shirt size without uploading anything

Close button is ok, just requires more customization than expected so I didn’t do it in the first place (low priority). Can add it now that all core functionalities are done.

File: src/core/components/profile/uploadPaymentBlock.component.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            try {
+              const result = await uploadDocument(formData)
+              setStatus(result.status)
+              setDisplayFile(result)
+              errorMessage.current = ""
+            } catch (error) {
+              if (!(error as AxiosError).request.responseText) {
+                errorMessage.current = "อัพโหลดไม่สำเร็จ กรุณาเลือกไฟล์ที่มีขนาดเล็กกว่านี้"
+              } else {
+                errorMessage.current = (error as AxiosError).request.responseText
```
Comment (miello) [inline src/core/components/profile/uploadPaymentBlock.component.tsx]: We can extract error message from response body instead

```suggestion
                errorMessage.current = (error as AxiosError).response?.data.message
```

File: src/modules/profile.module.tsx [hunk 3]
```diff
@@ -7,6 +7,10 @@ import RegisterCard from "../core/components/profile/registercard.component"
 import StepCardList from "../core/components/profile/stepCardList.component"
 import MyProfileModel from "../core/models/myprofile.models"
 import { FileStatus } from "../core/models/dto/application.dto"
+import CustomDialog from "../core/components/profile/customDialog.component"
```
Comment (saenyakorn) [inline src/modules/profile.module.tsx]: Apply lazy load here to reduce total blocking time

**Review by miello — APPROVED** [review]

**Review by saenyakorn — APPROVED** [review]

**Pull request merged**